### DISCUSSION TITLE
Overhauling Mastodon's Polling System

### DISCUSSION THREAD

**Discussion by SndChaser** [mastodon/mastodon#23365] [General]
I have a few thoughts on the current state of polls in Mastodon as the result of running several polls recently.  Some of these I believe are just adjustments to the defaults, a few of them would require a bit more work.

### Defaults changes:

- Allow more than 4 options.  Recommend 7. There's many instances where the poll creator will have 4-6 options, and need the default "other".  Trying to squeeze options into 4 entries can be confusing which can lead to less accurate results. (as an example  Github polls allow for up to 8 options.)

- Add more duration options.  I think 5 days would be a good middle ground between 3 and 7 days.  The last poll I ran was 7 days, and it seemed to be very active for the first 4-5 days, then dropped off.

### More complicated:

Multiple choice polls aren't as intuitive as they could be.  First, creating them isn't easily discovered.  Changing the option box shape isn't much feedback, and clicking the option box to change the mode isn't easily discovered.  From a poll participant standpoint it isn't all that obvious either: several people commented that they wished they could voted for more entries, even when the poll was multiple choice.

Also, multiple choice poll results aren't intuitive to viewers.  When viewers see percentages, they expect them to add up to %100. Obviously that won't be true with multiple choice polls (except in a small number of cases).  Additionally, I found that when I did the math on the results of a poll, the percentage answers were introducing some inaccuracy: I shouldn't get fractional results when I take the percentage of total voters for an entry.

This just seems to be a mess to me, instead I would suggest:

- Instead of displaying percentages, display the number of votes for each option in a multiple choice poll.
- Add a toggle to the poll creation form to indicate if the poll is single choice or multiple choice.
- Add a header to the user display of the poll that says "Choose an option" for single choice polls, and "Choose one or more" for multiple choice polls.

### The new feature suggestion:

Finally, someone else suggested in my last poll that it would be nice to have ranked polls. Basically allow people to rank the entries on a scale.  If there are three options, the range would be 1-3, if there are four options 1-4, etc.    The results would display the number of votes, and the average ranking of all the votes (ie, if there are 100 votes, tally all the entries for an option, and divide by 100).

I think in this case, the participants would have to rank all of the options that are presented in the poll.

So, these are the thoughts I have on the current polling system.  I thought I would start a discussion first, and then open enhancement issues if there is a consensus that it would be appropriate.

George

**Comment by SndChaser** [comment]
I see that issue #23295 has been created, and covers most of this discussion.  There are some things that I bring up here that aren't covered -- specifically the UI/UX points that I brought up in this item.

**Comment by grahamperrin** [comment]
Is it impossible to vote in one's own poll?

![image](https://github.com/user-attachments/assets/1a330bc3-ed66-46af-8255-19e7fa2d17ec)

**Reply by grahamperrin** [reply]
Apparently so. A 2019 comment: 

> … I'm not able to vote in my own polls …

- #10681

**Comment by louwers** [comment]
> Allow more than 4 options

I also think polls should allow only one option.
